### PR TITLE
Preps for 1.8.0-rc1

### PR DESCRIPTION
**Pull request by d3zd3z**
Update version fields for 1.8.0-rc1 release.

Signed-off-by: David Brown <david.brown@linaro.org>

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -3,6 +3,31 @@
...
+## Version 1.8.0
+
+The 1.8.0 release of MCUboot contains numerous fixes, and adds support
+for the NuttX RTOS, and the Expressif ESP32 SDK.
+
+### About this release
+
+- Add support for the NuttX RTOS.
+- Add support for the Espressif ESP32 SDK.
+- Numerous improvements to `boot_serial`.
```
Comment (nvlsianpu) [inline docs/release-notes.md:15]: encoding was reworked using cddl-gen, which removed dependency on tinycbor

File: docs/release-notes.md [hunk 2]
```diff
@@ -3,6 +3,31 @@
...
+- Add verious hooks to be able to change how image data is accessed.
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply docs/release-notes.md:16]: (currently covers only OVERWRITE_ONLY mode)

**Comment by nvlsianpu** [comment]
mbed-os port enabled direct-xip mode
extracted bootutil_public library which is common interface for mcuboot/application
Allow to boot primary image if secondary one is unreachable
Added AES256 image encryption support
Multimage boot for direct-xip and ram-load mode

**Review by nvlsianpu — APPROVED** [review]

**Review by utzig — APPROVED** [review]
LGTM

**Pull request merged**